src/lib/y2partitioner/widgets/overview.rb

```diff
@@ -67,15 +77,25 @@ def initialize(hostname)
         super(OverviewTree.new(items))
       end
 
-      # Pages whose cached content should be considered outdated
+      # Ensures that UIState clears obsolte statuses and is aware of current page
       #
-      # This is a hack introduced because the NFS page works in a completely
-      # different way in which triggering a full redraw every time something
-      # changes is not an option. This way, the NFS page can invalidate the
-      # cached contents of other pages supporting this mechanism.
+      # That's especially needed when initial page is a candidate of a no longer exist one
       #
-      # @return [Array<Symbol>] only :system supported so far
-      attr_accessor :invalidated_pages
+      # @see #initial_page
+      def init
+        super
+
+        UIState.instance.clear_dead_statuses(@pages.map(&:id))
+        UIState.instance.select_page(@current_page.tree_path)
```

Comment:
NP: thinking about the method name, what about a `UIState#prune` method with parameters?

~~~
def prune(keep: [])
  statuses.select! { |s| keep.include?(s.page_id) }
end
~~~

By default, `#prune` would delete everything, but we can use `:keep` parameter to filter out the pages to keep.

Anyway, this is a NP, I am also fine with current version.